src/http/body.rs

```diff
@@ -8,7 +8,31 @@ use tokio_core::io::{Codec, EasyBuf};
 
 use super::Chunk;
 
+pub struct Chunked {}
+
+impl Codec for Chunked {
+    type In = Chunk;
+    type Out = Chunk;
+
+    fn decode(&mut self, buf: &mut EasyBuf) -> io::Result<Option<Self::In>> {
+        let len = buf.len();
+        trace!("Chunked decode {} bytes", len);
+        let body = buf.drain_to(len);
+        Ok(
+            Some(
+                Chunk(Vec::from(body.as_ref()))
+            )
+        )
```

Comment:
Because we are using a pipelined implementation, we run the risk of having the end of a response being in the same buffer as the start of a new response.

Example buffer:

```
HTTP/1.1 200 OK
Content-Type: text/plain
Transfer-Encoding: chunked

7\r\n
Mozilla\r\n 
0\r\n 
\r\n
HTTP/1.1 200 OK
```

With this decoder we would send the start of the new response as part of the existing response.